sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4281,6 +4281,18 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         Row(2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22) :: Nil)
     }
   }
+
+  test("SPARK-38182: Fix NoSuchElementException if pushed filter does not contain any " +
+    "references") {
+    withTable("t") {
+      withSQLConf(SQLConf.OPTIMIZER_EXCLUDED_RULES.key ->
+        "org.apache.spark.sql.catalyst.optimizer.BooleanSimplification") {
+
+        sql("CREATE TABLE t (c1 int) USING PARQUET")
+        checkAnswer(sql("SELECT * FROM t WHERE c1 = 1 AND 2 > 1"), Nil)
```

Comment:
If we use `rand() > 0.5`, can we trigger the bug without disabling the optimizer rule?

Reply:
It can not trigger this bug since we can not push down indeterministic filter. see code in `FileSourceStrategy`

https://github.com/apache/spark/blob/e0bc977dd0dd378665da431d4a2497e05ae90cf7/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala#L157-L173